x/incentives/keeper/grpc_query.go

```diff
@@ -122,10 +126,16 @@ func (q Querier) ActiveGaugesPerDenom(goCtx context.Context, req *types.ActiveGa
 	valStore := prefix.NewStore(store, types.KeyPrefixActiveGauges)
 
 	pageRes, err := query.FilteredPaginate(valStore, req.Pagination, func(key []byte, value []byte, accumulate bool) (bool, error) {
-		activeGauges := q.Keeper.GetActiveGauges(ctx)
-		for _, gauge := range activeGauges {
-			if gauge.DistributeTo.Denom == req.Denom {
-				gauges = append(gauges, gauge)
+		newGauges, err := q.getGaugeFromIDJsonBytes(ctx, value)
+		if err != nil {
+			panic(err)
+		}
+		for i := 0; i < len(newGauges); i++ {
+			if newGauges[i].DistributeTo.Denom != req.Denom {
+				return false, nil
+			}
+			if accumulate {
+				gauges = append(gauges, newGauges...)
```

Comment:
Shouldn't we be appending a single element, i.e. `newGauges[i]` instead of the entire slice each time?

Reply:
yes sure, my mistake